front/pages/api/w/[wId]/vaults/[vId]/index.ts

```diff
@@ -0,0 +1,174 @@
+import type { VaultType, WithAPIErrorResponse } from "@dust-tt/types";
+import { PatchVaultRequestBodySchema } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { GroupResource } from "@app/lib/resources/group_resource";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import { apiError } from "@app/logger/withlogging";
+
+export type GetVaultResponseBody = {
+  vault: VaultType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetVaultResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+
+  if (
+    !vault ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "read"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
+      },
+    });
+  }
+
+  const group = await GroupResource.fetchByModelId(vault.groupId);
+  if (!group) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      res.status(200).json({ vault: vault.toJSON() });
+      return;
+    case "PATCH":
+      if (!auth.isAdmin() || !auth.isBuilder()) {
```

Comment:
So one need to either be an admin, or a builder that has access to the vault?

Reply:
yes